### PR TITLE
Add unit test suite, disable `test-checks`

### PR DESCRIPTION
**Pull request by vemv**
Reflects the decisions we made over https://github.com/clojure-emacs/clj-refactor.el/pull/493 (which I've postponed a bit).

Code has been copied as is, except for the unit tests in question, which are for `cljr--ns-name` (as a random example that I picked).

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -28,9 +28,14 @@ compile: elpa
...
 	rm -f $(OBJS)
 
-test: $(PKGDIR)
+integration-tests: $(PKGDIR)
 	$(CASK) exec ecukes --no-win
 
+unit-tests:
+	$(CASK) exec buttercup -L .
+
+test: unit-tests integration-tests
```
Comment (vemv) [inline Makefile:37]: refactoring based on https://github.com/clojure-emacs/clj-refactor.el/pull/493/files#r709812770

(@bbatsov)

File: test/test-checks.el [hunk 2]
```diff
@@ -15,16 +15,17 @@
 (setq checkdoc-arguments-in-order-flag nil)
 (setq checkdoc-verb-check-experimental-flag nil)
 
-(let ((files (directory-files default-directory t
-                              "\\`[^.].*\\.el\\'" t)))
+(when nil ;; See https://github.com/clojure-emacs/clj-refactor.el/issues/491
```
Comment (bbatsov) [inline test/test-checks.el:18]: I think you can also set some env var and just disable this on the CI. I think it's not a bad idea for people to be able to run this locally.

File: tests/unit-test.el [hunk 3]
```diff
@@ -0,0 +1,9 @@
+(require 'paredit)
+(require 'clj-refactor)
+
+;; NOTE: please remember, without an `it` block, your tests will not be run!
```
Comment (bbatsov) [inline tests/unit-test.el:4]: You also link to buttercup's docs here.

**Comment by bbatsov** [comment]
Looks good overall, feel free to ignore my remarks.

**Comment by vemv** [comment]
Amended w/ feedback, thanks!

**Pull request merged**